app/models/user.rb

```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+    # Validation for unique and valid email addres
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+    validates :email, uniqueness: { case_sensitive: false }, format: { with: VALID_EMAIL_REGEX }
```

Comment:
👍 Props for disable case sensitive

Reply:
Does it allow to have blank (nil or '') email?

Reply:
No it does not.